packages/text-input/src/TextInput.tsx

```diff
@@ -0,0 +1,354 @@
+ * @param props.label Text shown in bold above the input element.
+ * @param props.description Text that gives more detail about the requirements for the input.
+ * @param props.optional Whether or not the field is optional.
+ * @param props.disabled Whether or not the field is currently disabled.
+ * @param props.onChange Callback to be executed when the value of the input field changes.
+ * @param props.placeholder The placeholder text shown in the input field before the user begins typing.
+ * @param props.errorMessage The message shown below the input field if the value is invalid.
+ * @param props.state The current state of the TextInput. This can be none, valid, or error.
+ * @param props.value The current value of the input field. If a value is passed to this prop, component will be controlled by consumer.
+ * @param props.className className supplied to the TextInput container.
+ */
+const TextInput = React.forwardRef(
+  (
+    {
+      id: propsId,
+      label,
+      description,
+      optional = false,
+      disabled = false,
+      onChange,
+      placeholder,
+      errorMessage,
+      state,
+      value: controlledValue,
+      className,
+    }: TextInputProps,
+    forwardRef: React.Ref<HTMLInputElement>,
+  ) => {
+    const isControlled = typeof controlledValue === 'string';
+    const [uncontrolledValue, setValue] = useState('');
+    const value = isControlled ? controlledValue : uncontrolledValue;
+    const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+    const [hasFocus, setHasFocus] = useState(false);
+    const inputSelectorProp = createDataProp('input-selector');
+    const [generatedId] = useState(
+      `text-input-${Math.floor(Math.random() * 10000000)}`,
+    );
+    const id = propsId ?? generatedId;
+
+    function getInputColorFromState() {
+      if (state === State.Error) {
+        return uiColors.red.base;
+      }
+
+      if (state === State.Valid) {
+        return uiColors.green.base;
+      }
+
+      return '#CCC';
+    }
+
+    function getInputPaddingFromState() {
+      if (state === State.Valid || state === State.Error) {
+        return '30px';
+      }
+
+      if (optional) {
+        return '60px';
+      }
+
+      return '12px';
+    }
+
+    function onValueChange(e: React.ChangeEvent<HTMLInputElement>) {
+      if (onChange) {
+        onChange(e);
+      }
+
+      if (!isControlled) {
+        setValue(e.target.value);
+      }
+    }
+
+    const interactionRingFocusStyle = css`
+      ${inputSelectorProp.selector}:focus ~ & {
+        background-color: #9dd0e7;
+        transform: scale(1);
+        z-index: 1;
+      }
+    `;
+
+    const interactionRingHoverStyle = css`
+      ${inputSelectorProp.selector}:hover ~ & {
+        transform: scale(1);
+      }
+    `;
+
+    const descriptionStyle = css`
+      color: ${uiColors.gray.dark1};
+      font-size: 14px;
+      height: ${description === '' ? '0' : '20'}px;
```

Comment:
Rather than doing this, let's not render the paragraph tag for the description at all when there isn't one.

Reply:
Fixed